frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -303,9 +303,32 @@ export function LinkWithTargetBlank(props: LinkProps): ReactElement {
   )
 }
 
+function fixBodyForHeader(tag: string, body: string): string {
+  switch (tag) {
```

Comment:
Maybe we should do `tag.toLowerCase()`? It's not a big deal, but since the proto will be a string and not an enum (we probably should have thought more on that), doing this would just handle all cases.